aliasmgr/aliasmgr_test.go

```diff
@@ -266,3 +267,234 @@ func TestGetNextScid(t *testing.T) {
+// loaded correctly.
+func TestBackwardCompatibility(t *testing.T) {
+	t.Parallel()
+
+	// Create the backend database and use this to create the aliasStore.
+	dbPath := filepath.Join(t.TempDir(), "testdb")
+	db, err := kvdb.Create(
+		kvdb.BoltBackendName, dbPath, true, kvdb.DefaultDBTimeout,
+		false,
+	)
+	require.NoError(t, err)
+	defer db.Close()
+
+	linkUpdater := func(shortID lnwire.ShortChannelID) error {
+		return nil
+	}
+
+	// Manually write an old-format alias (8 bytes, no flags).
+	const (
+		base  = uint64(111111111)
+		alias = uint64(222222222)
+	)
+
+	err = kvdb.Update(db, func(tx kvdb.RwTx) error {
+		bucket, err := tx.CreateTopLevelBucket(aliasBucket)
+		if err != nil {
+			return err
+		}
+
+		var aliasBytes [8]byte
+		var baseBytes [8]byte
+		byteOrder.PutUint64(aliasBytes[:], alias)
+		byteOrder.PutUint64(baseBytes[:], base)
+
+		// Write only 8 bytes (old format, no flags).
+		return bucket.Put(aliasBytes[:], baseBytes[:])
+	}, func() {})
+	require.NoError(t, err)
+
+	// Create the manager - it should load the old entry.
+	aliasStore, err := NewManager(db, linkUpdater)
+	require.NoError(t, err)
+
+	baseScid := lnwire.NewShortChanIDFromInt(base)
+	aliasScid := lnwire.NewShortChanIDFromInt(alias)
+
+	// The alias should be in baseToSet.
+	aliases := aliasStore.GetAliases(baseScid)
+	require.Len(t, aliases, 1)
+	require.Contains(t, aliases, aliasScid)
+
+	// The alias should be findable (since it's not confirmed).
+	foundBase, err := aliasStore.FindBaseSCID(aliasScid)
+	require.NoError(t, err)
+	require.Equal(t, baseScid, foundBase)
+
+	// Mark as confirmed - old aliases should not persist.
+	err = aliasStore.DeleteSixConfs(baseScid)
+	require.NoError(t, err)
+
+	// Should no longer be findable after confirmation.
+	_, err = aliasStore.FindBaseSCID(aliasScid)
+	require.Error(t, err)
+}
+
+// TestDeletePersistentAlias tests that persistent aliases can be manually
+// deleted via DeleteLocalAlias.
+func TestDeletePersistentAlias(t *testing.T) {
+	t.Parallel()
+
+	// Create the backend database and use this to create the aliasStore.
+	dbPath := filepath.Join(t.TempDir(), "testdb")
+	db, err := kvdb.Create(
+		kvdb.BoltBackendName, dbPath, true, kvdb.DefaultDBTimeout,
+		false,
+	)
+	require.NoError(t, err)
+	defer db.Close()
+
+	updateChan := make(chan struct{}, 1)
+	linkUpdater := func(shortID lnwire.ShortChannelID) error {
+		updateChan <- struct{}{}
+		return nil
+	}
+
+	aliasStore, err := NewManager(db, linkUpdater)
+	require.NoError(t, err)
+
+	const (
+		base            = uint64(123123123)
+		persistentAlias = uint64(789789789)
+	)
+
+	baseScid := lnwire.NewShortChanIDFromInt(base)
+	persistentScid := lnwire.NewShortChanIDFromInt(persistentAlias)
+
+	// Add a persistent alias (with WithBaseLookup option).
+	err = aliasStore.AddLocalAlias(
+		persistentScid, baseScid, true, true, WithBaseLookup(),
+	)
+	require.NoError(t, err)
+
+	// Link updater should be called.
+	<-updateChan
```

Comment:
Should select with a timeout here to avoid the test blocking for up to 10 mins in case of a failure.

Reply:
updated all sites similar to this, even for previous tests